lib/smart_name.rb

```diff
@@ -112,186 +125,22 @@ def == other
-          junction_pieces << piece
-          piece
-        end
-        parts + junction_pieces
-      end
-  end
-
-  # ~~~~~~~~~~~~~~~~~~~~ SHOW / ABSOLUTE ~~~~~~~~~~~~~~~~~~~~
-
-  def to_show *ignore
-    ignore.map!(&:to_name)
-
-    show_parts = parts.map do |part|
-      reject = (part.empty? || (part =~ /^_/) || ignore.member?(part.to_name))
-      reject ? nil : part
-    end
-
-    show_name = show_parts.compact.to_name.s
-
-    case
-    when show_parts.compact.empty? then  self
-    when show_parts[0].nil?        then  self.class.joint + show_name
-    else show_name
-    end
-  end
-
-  def to_absolute context, args={}
-    context = context.to_name
-    parts.map do |part|
-      new_part =
-        case part
-        when /^_user$/i
-          name_proc = self.class.session
-          name_proc ? name_proc.call : part
-        when /^_main$/i            then self.class.params[:main_name]
-        when /^(_self|_whole|_)$/i then context.s
-        when /^_left$/i            then context.trunk
-          # note - inconsistent use of left v. trunk
-        when /^_right$/i           then context.tag
-        when /^_(\d+)$/i
-          pos = $~[1].to_i
-          pos = context.length if pos > context.length
-          context.parts[pos - 1]
-        when /^_(L*)(R?)$/i
-          l_s, r_s = $~[1].size, !$~[2].empty?
-          l_part = context.nth_left l_s
-          r_s ? l_part.tag : l_part.s
-        # when /^_/
-        #   custom = args[:params] ? args[:params][part] : nil
-        #   custom ? CGI.escapeHTML(custom) : part
-        # why are we escaping HTML here?
-        else
-          part
-        end.to_s.strip
-      new_part.empty? ? context.to_s : new_part
-    end * self.class.joint
-  end
-
-  def to_absolute_name *args
-    self.class.new to_absolute(*args)
-  end
-
-  def nth_left n
-    # 1 = left; 2= left of left; 3 = left of left of left....
-    (n >= length ? parts[0] : parts[0..-n - 1]).to_name
+  # @return true if name has a chain of parts that equals `subname`
+  def include? subname
+    subkey = subname.to_name.key
+    joint = Regexp.quote self.class.joint
+    key =~ /(^|#{joint})#{Regexp.quote subkey}($|#{joint})/
   end
 
   # ~~~~~~~~~~~~~~~~~~~~ MISC ~~~~~~~~~~~~~~~~~~~~
 
-  def replace_part oldpart, newpart
-    oldpart = oldpart.to_name
-    newpart = newpart.to_name
-    if oldpart.simple?
-      if simple?
-        self == oldpart ? newpart : self
-      else
-        parts.map do |p|
-          oldpart == p ? newpart.to_s : p
-        end.to_name
-      end
-    elsif simple?
-      self
-    else
-      if oldpart == parts[0, oldpart.length]
-        if length == oldpart.length
-          newpart
-        else
-          (newpart.parts + parts[oldpart.length..-1]).to_name
-        end
-      else
-        self
-      end
-    end
-  end
-
   # HACK. This doesn't belong here.
   # shouldn't it use inclusions???
   def self.substitute! str, hash
```

Comment:
What is this supposed to do? Does "inclusions" refer to wagn nests? I don't see it used anywhere, neither in smartname nor in wagn.